### DISCUSSION TITLE
"Examples" of command

### DISCUSSION THREAD

**Discussion by wayneyaoo** [Tyrrrz/CliFx#114] [General]
Can we have concept of "Example" for each command that can be specified by the CLI author and a new section is rendered as "EXAMPLE".

I was imaging something like
```C#
[Command("verb")]
public class MyCommand : ICommand
{
    // options and parameters go here as property

    [Example]
    public (static) Example[] Examples = new [] 
        { new Example("verb --option value"), new Example("verb --anotherPositionalOption") };
}
``` 

and the examples could be rendered to another section like "EXAMPLE" when `./myapp verb -h` is called.

I'm happy to contribute if this looks good. Or we can discuss.

**Comment by Tyrrrz** [comment]
Hi.
Could also just be a parameter on `[Command]` instead, right? Something like:

```csharp
[Command("verb", Examples = new[]
{
    "verb --option value",
    "verb --anotherPositionalOption"
})]
public class MyCommand : ICommand
```

Maybe even skip the `verb` because it can inferred 🤔

**Reply by Tyrrrz** [reply]
PS: I moved this into an issue: #115
